src/anomalib/metrics/pimo/functional.py

```diff
@@ -276,6 +275,73 @@ def aupimo_scores(
 # =========================================== AUX ===========================================
 
 
+def _binary_search_threshold_at_fpr_target(
+    anomaly_maps_normals: torch.Tensor,
+    fpr_target: float | torch.Tensor,
+    maximum_iterations: int = 300,
+) -> float:
+    """Binary search of threshold that achieves the given shared FPR level.
```

Comment:
It would be good to add a more detailed description to this docstring, explaining why we need to apply the binary search and how it is performed. This would be useful for future reference.